src/bors/context.rs

```diff
@@ -1,13 +1,32 @@
-use crate::{bors::command::CommandParser, database::DbClient};
-use std::sync::Arc;
+use arc_swap::ArcSwap;
 
-pub struct BorsContext {
+use crate::{bors::command::CommandParser, database::DbClient, github::GithubRepoName};
+use std::{collections::HashMap, sync::Arc};
+
+use super::{GlobalClient, RepositoryClient, RepositoryState};
+
+pub struct BorsContext<Client: RepositoryClient> {
     pub parser: CommandParser,
     pub db: Arc<dyn DbClient>,
+    pub global_client: Arc<dyn GlobalClient<Client>>,
+    pub repositories: Arc<ArcSwap<HashMap<GithubRepoName, Arc<RepositoryState<Client>>>>>,
 }
 
-impl BorsContext {
-    pub fn new(parser: CommandParser, db: Arc<dyn DbClient>) -> Self {
-        Self { parser, db }
+impl<Client: RepositoryClient> BorsContext<Client> {
+    pub async fn new(
+        parser: CommandParser,
+        db: Arc<dyn DbClient>,
+        global_client: Arc<dyn GlobalClient<Client>>,
+    ) -> Self {
+        // this unwrap is making me nervous, but if lhe repos loading
```

Comment:
We can just return `anyhow::Result<Self>` from the constructor, no need to `unwrap` here.